src/node/history.rs

```diff
@@ -0,0 +1,79 @@
+#[cfg(test)]
+mod tests;
```

Comment:
I *really* do not want to have unit tests in the same file as the code, because it hurts readability. Unfortunately this forces me to structure these files is an annoying way. For each module I end up with a directory containing a single file:
```
src/
├── node/
│   ├── history.rs
│   ├── history/
│   │   └── tests.rs
│   ├── mempool.rs
│   ├── mempool/
│   │   └── tests.rs
```

If I put the test module anywhere else, it will not be able to access private stuff in the tested module.

So far I did not find a good workaround.